File: source/nestjs-petclinic-rest/src/services/dto/owner.dto.ts
import { PetDTO } from './pet.dto';
import { Owner } from './../../domain/entities/owner.entity';
import { ApiProperty } from '@nestjs/swagger';

export class OwnerDTO
{
  @ApiProperty()
  id: number;

  @ApiProperty()
  firstName: string;

  @ApiProperty()
  lastName: string;

  @ApiProperty()
  address: string;

  @ApiProperty()
  city: string;

  @ApiProperty()
  telephone: string;

  @ApiProperty()
  pets: PetDTO[];

  static fromOwner(owner: Owner) {
    if (owner == null) {
        return null;
    }
    const dto = new OwnerDTO();
    dto.id = owner.id;
    dto.firstName = owner.firstName;
    dto.lastName = owner.lastName;
    dto.address = owner.address;
    dto.city = owner.city;
    dto.telephone = owner.telephone;
    dto.pets = owner.pets?.map(x => PetDTO.fromPet(x));
    return dto;
  }

  static requiredRelations: string[] = ["pets", "pets.petType", "pets.owner", "pets.visits"];
}

File: source/dotnet-cqrs-petclinic-rest/PetClinic.Application/Owners/OwnerDto.cs
using System;
using System.Collections.Generic;
using AutoMapper;
using Intent.RoslynWeaver.Attributes;
using PetClinic.Application.Common.Mappings;
using PetClinic.Application.Pets;
using PetClinic.Domain.Entities;

[assembly: DefaultIntentManaged(Mode.Fully)]
[assembly: IntentTemplate("Intent.Application.Dtos.DtoModel", Version = "1.0")]

namespace PetClinic.Application.Owners
{
    public class OwnerDto : IMapFrom<Owner>
    {
        public OwnerDto()
        {
            FirstName = null!;
            LastName = null!;
            Address = null!;
            City = null!;
            Telephone = null!;
            Pets = null!;
        }

        public int Id { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string Address { get; set; }
        public string City { get; set; }
        public string Telephone { get; set; }
        public List<PetDto> Pets { get; set; }

        public static OwnerDto Create(
            int id,
            string firstName,
            string lastName,
            string address,
            string city,
            string telephone,
            List<PetDto> pets)
        {
            return new OwnerDto
            {
                Id = id,
                FirstName = firstName,
                LastName = lastName,
                Address = address,
                City = city,
                Telephone = telephone,
                Pets = pets
            };
        }

        public void Mapping(Profile profile)
        {
            profile.CreateMap<Owner, OwnerDto>()
                .ForMember(d => d.Pets, opt => opt.MapFrom(src => src.Pets));
        }
    }
}